### PR TITLE
[VFX] Add Performance Test

### PR DESCRIPTION
**Pull request by PaulDemeulenaere**
### Add performance test for VFX

For VFX_URP & VFX_HDRP :
- PlayMode & Standalone : Memory & Counter (CPU/GPU) for every graphicTest
- EditMode :
   - Load VFX Library (which correspond to the first listing of vfx model)
   - Backup And Restore : Worst case for Undo/Redo operation, rebuild all the controller from scratch.
   - Compilation : shaderValidation on/off & mode edition/runtime.
   - VFXViewWindow Open And Render : Open every vfx asset & render a few frame moving the view to force repaint.

**See** : https://favro.com/organization/c564ede4ed3337f7b17986b6/4ab0cf9c9a0d6d61242bd015?card=Uni-131889

_performance_test

**Note :**
1. There are a couple of fix in `com.unity.testing.graphics-performance` which should be applied to https://github.cds.internal.unity3d.com/unity/com.unity.testing.graphics-performance, @alelievr is also concerned by this kind of change.

2. This branch should be merge also in private : https://github.cds.internal.unity3d.com/unity/ScriptableRenderPipelinePrivate/pull/54

**Status :**
_Profile test :_
- [x] UI render analysis
- [x] Compilation
- [x] Runtime execution

_TODO :_
- [ ] Configure Yamato
   - [x] Add PerformanceTest target
   - [ ] Add All PerformanceTest target
- [ ] Configure Grafana

### CODE REVIEW

**Comment by PaulDemeulenaere** [comment]
Awaiting for this https://github.com/Unity-Technologies/Graphics/pull/2987 & https://github.cds.internal.unity3d.com/unity/ScriptableRenderPipelinePrivate/pull/44

**Review by theo-pnv — APPROVED** [review]
Yamato-wise this looks good 👍🏻

**Comment by PaulDemeulenaere** [comment]
> Yamato-wise this looks good 👍🏻

There is still an issue on private repository, I'm searching for the reason it fails only with these public change : 
https://yamato.cds.internal.unity3d.com/jobs/339-ScriptableRenderPipelinePrivate/tree/vfx%252Ftest%252Fadd-performance-test-without-yamato-change

**Comment by PaulDemeulenaere** [comment]
See

> > Yamato-wise this looks good 👍🏻
> 
> There is still an issue on private repository, I'm searching for the reason it fails only with these public change :
> https://yamato.cds.internal.unity3d.com/jobs/339-ScriptableRenderPipelinePrivate/tree/vfx%252Ftest%252Fadd-performance-test-without-yamato-change

[Fixed ](https://yamato.cds.internal.unity3d.com/jobs/339-ScriptableRenderPipelinePrivate/tree/vfx%252Ftest%252Fadd-performance-test-without-yamato-change): this change only targets this PR, the actual [configuration PR](https://github.cds.internal.unity3d.com/unity/ScriptableRenderPipelinePrivate/pull/54) can be merged afterwards. 🎉

File: com.unity.testing.visualeffectgraph/PerformanceTests/Runtime/VFXPerformanceRuntimeTests.cs [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+                yield return "VFXEditor.VisualEffectImporter.GenerateAssetDataOneShader";
+                yield return "VFXEditor.VisualEffectResource.GetResourceAtPath";
+                yield return "VFXEditor.VisualEffectResource.GetResourceAtPath_Depoint";
+            }
+        }
+
+        public static IEnumerator Load_And_Prepare(GraphicsTestCase testCase)
+        {
+            UnityEngine.SceneManagement.SceneManager.LoadScene(testCase.ScenePath);
+            yield return null;
```
Comment (PaulDemeulenaere) [inline com.unity.testing.visualeffectgraph/PerformanceTests/Runtime/VFXPerformanceRuntimeTests.cs:85]: Replace `yield return null;` by `yield return new WaitForEndOfFrame()` to always process a frame.

**Review by alelievr — APPROVED** [review]

**Comment by PaulDemeulenaere** [comment]
Last verification check, we improved memory test :
- https://yamato.cds.internal.unity3d.com/jobs/339-ScriptableRenderPipelinePrivate/tree/vfx%252Ftest%252Fadd-performance-test
- https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/vfx%252Ftest%252Fadd-performance-test
⏳

**Comment by PaulDemeulenaere** [comment]
Got a time out while running a test :
```
[22:42:40.467 INF] [41/261] TestFinish
	name: UnityEditor.VFX.PerformanceTest.VFXRuntimeMemoryTests.HDRP_DebugAlbedo
	result: FAILED
	message: Test exceeded Timeout value of 120000ms
	duration: 137.830452 seconds
```
And we are lacking of test filter in PlayMode XR.

**Comment by PaulDemeulenaere** [comment]
> Got a time out while running a test :
> 
> ```
> [22:42:40.467 INF] [41/261] TestFinish
> 	name: UnityEditor.VFX.PerformanceTest.VFXRuntimeMemoryTests.HDRP_DebugAlbedo
> 	result: FAILED
> 	message: Test exceeded Timeout value of 120000ms
> 	duration: 137.830452 seconds
> ```
> 
> And we are lacking of test filter in PlayMode XR.

Resolve at b05479525445ad6fb10327f3a9de6cf25671c5ad Thanks @theo-pnv

**Comment by PaulDemeulenaere** [comment]
Waiting for internal result, see this [PR](https://github.cds.internal.unity3d.com/unity/ScriptableRenderPipelinePrivate/pull/54) ⏳

**Comment by PaulDemeulenaere** [comment]
https://github.com/Unity-Technologies/Graphics/pull/203#issuecomment-763604706
The PR in internal is also ready, merging both.

**Pull request merged**